pkg/signup/signup_service.go

```diff
@@ -126,8 +128,8 @@ func (s *ServiceImpl) GetSignup(userID string) (*Signup, error) {
 	}
 
 	// Check UserSignup status to determine whether user signup is complete
-	signupCondition, complete := condition.FindConditionByType(userSignup.Status.Conditions, crtapi.UserSignupComplete)
-	if !complete {
+	signupCondition, found := condition.FindConditionByType(userSignup.Status.Conditions, crtapi.UserSignupComplete)
+	if !found || (found && signupCondition.Status != apiv1.ConditionTrue) {
```

Comment:
I think something like `condition.IsTrue()` would be helpful here. I've created a PR for common - https://github.com/codeready-toolchain/toolchain-common/pull/45

Reply:
I just refactored this code a little because I also have to handle the scenario where the condition does not even exist.  I've added a test for this also.  I'm not sure a `condition.IsTrue()` would be helpful in this case.

Reply:
> I'm not sure a condition.IsTrue() would be helpful in this case.

Why not?
https://github.com/codeready-toolchain/toolchain-common/pull/45/files#diff-c3466dd19dde2ab03d3be49a50141b77R162

Reply:
Just for clarification. You change is correct. And it's not about the case when the condition doesn't exist only. The original code was incorrect. It treated a found condition (even if it's set to False!) as True condition. It was a bug.
So, basically my PR just moves this functionality to common to `condition.IsTrue()` helper function.

Reply:
So, when the PR is merged you could use it:
```
if condition.IsTrue(userSignup.Status.Conditions, crtapi.UserSignupComplete) {
    ...
}
```

Reply:
You're right, the original code was incorrect which is why I changed it.  I still can't use `condition.IsTrue()` because I also need to know whether the actual condition exists or not.  If the condition exists, I populate the `Signup` object with the `Reason` and `Message` values from the condition, otherwise I set them to hard-coded values.

Reply:
Ah, right. You need the condition itself to get its message and reason.